### PR TITLE
`InsertResult` to return the primary key's type

### LINKED ISSUE #103: `InsertResult` to return the primary key's type
Currently, the `last_insert_id` from InsertResult simply uses the ExecResult's last_insert_id.
However, at least for Postgres with `returning`, we should be able to retrieve the primary key column for the Entity and return the value in its native type. For example, a `uuid` instead of an `i64`.

### PR DESCRIPTION
**Pull request by billy1624**
Resolve #103

Special thanks to @Acidic9 for providing inspiration on PR #112

### CODE REVIEW

File: src/executor/execute.rs [hunk 1]
```diff
@@ -21,23 +21,36 @@ pub(crate) enum ExecResultHolder {
 // ExecResult //
 
 impl ExecResult {
```
Comment (tyt2y3) [inline src/executor/execute.rs:23]: Is there a way we can keep `ExecResult` as is? It is just a wrapper over SQLx's QueryResult. We should not be changing ExecResult. We should only change InsertResult.

Comment (billy1624) [reply src/executor/execute.rs:23]: Let me think...

File: src/executor/insert.rs [hunk 2]
```diff
@@ -83,15 +83,21 @@ where
...
             let res = conn.query_one(statement).await?.unwrap();
-            crate::query_result_into_exec_result(res)?
+            res.try_get("", "last_insert_id").unwrap_or_default()
         }
-        _ => db.execute(statement).await?,
+        _ => {
+            db.execute(statement).await?
+            .last_insert_id()
+            .to_string()
+            .parse()
```
Comment (tyt2y3) [inline src/executor/insert.rs:96]: Yeah this looks much more sensible. May I also know why is it necessary to `to_string` and `parse` again?

Comment (billy1624) [reply src/executor/insert.rs:96]: Because we cannot convert e.g. `u64` to `i32` directly

**Comment by billy1624** [comment]
If this passes all tests, I will start re-think about all these trait bounds

https://github.com/SeaQL/sea-orm/blob/4e5a8d2dc553a26668c89b5740bb6a0d4b5e33f3/src/entity/primary_key.rs#L10-L18

File: src/executor/insert.rs (hunk 2 above)
Comment (tyt2y3) [reply src/executor/insert.rs:96]: how about this https://stackoverflow.com/questions/28273169/how-do-i-convert-between-numeric-types-safely-and-idiomatically?

**Comment by billy1624** [comment]
Why I got this error again?
https://github.com/SeaQL/sea-orm/runs/3454855146#step:6:104

File: src/executor/insert.rs (hunk 2 above)
Comment (billy1624) [reply src/executor/insert.rs:96]: I saw this StackOverflow this morning. I don't think it helps :(

Comment (tyt2y3) [reply src/executor/insert.rs:96]: Why not?

Comment (billy1624) [reply src/executor/insert.rs:96]: We can't use `TryFrom` or `From` for it

Comment (tyt2y3) [reply src/executor/insert.rs:96]: Why not?

Comment (billy1624) [reply src/executor/insert.rs:97]: If we want `TryFrom` or `From` `u32` to `ValueType`, this requires `ValueType: From<u32>`. Which is not reasonable

Comment (tyt2y3) [reply src/executor/insert.rs:97]: But having to stringify and parse is not much more sensible either

**Comment by billy1624** [comment]
> Why I got this error again?
> https://github.com/SeaQL/sea-orm/runs/3454985766?check_suite_focus=true#step:6:104

Okay, solve it on 4d0c5f0

File: src/executor/insert.rs (hunk 2 above)
Comment (billy1624) [reply src/executor/insert.rs:97]: Agree

**Comment by tyt2y3** [comment]
We need a trait `TryFromU64` to replace the use of `FromStr`

**Comment by Acidic9** [comment]
> We need a trait `TryFromU64` to replace the use of `FromStr`

How would TryFromU64 work with something like UUID? I don't think postgres returns UUID as an integer does it?

**Comment by tyt2y3** [comment]
I think the Postgres side will fairly ignore that result. We were only trying to fix the MySQL side.

**Comment by billy1624** [comment]
Wow the commit hash starts with 0e0ee0e lolll

**Comment by Acidic9** [comment]
I'm curious why the trait TryFromU64 is added instead of using `From<u64>` 🤔

**Comment by billy1624** [comment]
> I'm curious why the trait `TryFromU64` is added instead of using `From<u64>` or `TryFrom<u64>` 🤔
> It seems like this might cover cases for libraries with types that already implement `From<u64>` / `TryFrom<u64>`
> Sorry if this has already been discussed

I think what we have here is not an usual logic of `From<u64>` or `TryFrom<u64>`. So, we better not use the conventional std traits.

File: src/executor/query.rs [hunk 3]
```diff
@@ -280,3 +288,135 @@ impl TryGetable for Decimal {
...
+            cols.len()
+        ))))
+    } else {
+        Ok(())
+    }
+}
+
+// TryFromU64 //
+
+pub trait TryFromU64: Sized {
```
Comment (tyt2y3) [inline src/executor/query.rs:349]: Awesome!

File: src/tests_cfg/cake_filling_price.rs [hunk 4]
```diff
@@ -35,6 +35,8 @@ pub enum PrimaryKey {
 }
 
 impl PrimaryKeyTrait for PrimaryKey {
+    type ValueType = (i32, i32);
```
Comment (tyt2y3) [inline src/tests_cfg/cake_filling_price.rs:38]: Seems like we are able to support composite key now!

Comment (billy1624) [reply src/tests_cfg/cake_filling_price.rs:38]: Indeed

**Pull request merged**